packages/gatsby/src/bootstrap/remove-stale-jobs.js

```diff
@@ -0,0 +1,25 @@
+const { isJobStale } = require(`../utils/jobs-manager`)
+const { publicActions, internalActions } = require(`../redux/actions`)
+
+module.exports = state => {
+  const actions = []
+
+  // If any of our finished jobs are stale we remove them to keep our cache small
+  state.jobsV2.done.forEach((job, contentDigest) => {
+    if (isJobStale(job)) {
+      actions.push(internalActions.removeStaleJob(contentDigest))
+    }
+  })
```

Comment:
This can be very expensive for large caches as it will `fs.exists` on whole lot of files there. Is it actually needed?

Reply:
if we don't, our cache might keep growing as we save all done jobs. Unsure how to make this more scalable.. We could leverage a touchJob function that acts the same as touchNode to keep results alive

Reply:
I don't have any currently and maybe this is just something that we can't work around.

But also with this, it doesn't necessarily remove all trully stale jobs (i.e. if we don't remove input file, and change image transformation args - we will keep both old jobs and add new ones). I do think we need something more to figure out which jobs are trully stale?

Reply:
I know we do but it's a good start, eventually, you want to have something own a job so when it gets removed we know the job needs to go. When a plugin gets removed we still keep the jobs alive.

Mostly this is tied to graphql nodes....